ftdc/ftdc.go

```diff
@@ -174,7 +184,8 @@ func (ftdc *FTDC) Remove(name string) {
 		"name", name, "generationId", ftdc.inputGenerationID)
 }
 
-// Start spins off the background goroutine for collecting/writing FTDC data.
+// Start spins off the background goroutine for collecting + writing FTDC data. It's normal for tests
```

Comment:
Sorry for the extra comments. Feel free to skip. Or not.

As I was mucking with out `getWriter` worked and manipulated variables, I needed to convince myself we didn't need locks anywhere (for neither production nor, more importantly, testing). This documentation was the result of that critical thinking.

Particularly the relationship between the `statsWriter`, `writeDatum` and `getWriter`. I concluded it never makes sense for an application to call any of those concurrently.

Which also means if a caller spins up a goroutine for the `statsWriter`, it's totally illegal to use `writeDatum`/`getWriter`. One may only communicate at that point with the `datumCh`.

But if a caller does not spin off the `statsWriter`, they may interleave `writeDatum` and `getWriter` as they please.

Reply:
Particularly because this is internal/hopefully-untouched-except-by-you-and-netcode API, I see no reason to add concurrency primitives to _stop_ something like `statsWriter` + concurrent `writeDatum` + `getWriter` from happening. I.e. your reasoning makes sense to me, but I think the comments you have left are important for posterity.